react/features/video-quality/subscriber.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+
+import debounce from 'lodash/debounce';
+
+import { _handleParticipantError } from '../base/conference';
+import { StateListenerRegistry } from '../base/redux';
+import { reportError } from '../base/util';
+
+import { VIDEO_QUALITY_LEVELS } from './constants';
+
+declare var APP: Object;
+
```

Comment:
Shouldn't we move the subscriber from the middleware too?

Reply:
Yes, I missed those. Thanks for pointing it out.